src/OpenApiLinter.php

```diff
@@ -0,0 +1,161 @@
+    $options = array(
+      'openapi-spec.config' => array(
+        'type' => 'optional string',
+        'help' => pht('Config file that defines the validation rules (https://github.com/IBM/openapi-validator#configuration-file)'),
+      ),
+      'openapi-spec.errors_only' => array(
+        'type' => 'optional bool',
+        'help' => pht('Only print the errors, ignore the warnings'),
+      ),
+    );
+    return $options + parent::getLinterConfigurationOptions();
+  }
+
+  public function setLinterConfigurationValue($key, $value) {
+    switch ($key) {
+      case 'openapi-spec.config':
+        $this->config = $value;
+        return;
+      case 'openapi-spec.errors_only':
+        $this->errors_only = $value;
+        return;
+    }
+    return parent::setLinterConfigurationValue($key, $value);
+  }
+
+  public function getNodeBinary() {
+    return 'lint-openapi';
+  }
+
+  protected function getMandatoryFlags() {
+    return array('--json');
+  }
+
+  protected function getDefaultFlags() {
+    $flags = array();
+    if ($this->config) {
+      $flags[] = '--config='.$this->config;
+    }
+    if ($this->errors_only) {
+      $flags[] = '--errors_only';
+    }
+    return $flags;
+  }
+
+  protected function parseLinterOutput($path, $err, $stdout, $stderr) {
+    $messages = array();
+
+    if ($err) {
+      throw new Exception(pht("While reading %s, an error occurred:\n%s", $path, $stderr));
+    }
+    /* Sample Output
+    {
+      "errors": {
+        "spectral": [
+          {
+            "path": [
+              "servers",
+              "0",
+              "url"
+            ],
+            "message": "Server URL should not have a trailing slash.",
+            "line": 9
+          },
+        ]
+      },
+      "warnings": {
+        "spectral": [
+          {
+            "path": [],
+            "message": "OpenAPI object should have non-empty `tags` array.",
+            "line": 0
+          }
+         ]
+      },
+      "error": true,
+      "warning": true
+    }
+    */
+    $json = json_decode($stdout, true);
+    if (!is_array($json)) {
+      throw new Exception(pht("While reading %s, an error occurred:\n%s", $path, $stderr));
+    }
+    if (array_key_exists('errors', $json)) {
+      $errors = $json['errors'];
+      $messages = $this->processOutput($path, $errors, ArcanistLintSeverity::SEVERITY_ERROR);
+    }
+    if (array_key_exists('warnings', $json)) {
+      $warnings = $json['warnings'];
+      $messages += $this->processOutput($path, $warnings, ArcanistLintSeverity::SEVERITY_WARNING);
+    }
+    return $messages;
+  }
+
+  protected function processOutput($path, $outputCategories, $severity) {
+    $messages = array();
+    foreach ($outputCategories as $outputCategory) {
+      foreach ($outputCategory as $output) {
+        $message = new ArcanistLintMessage();
+        $message->setPath($path)
+          ->setCode(is_array($output['path']) ? implode('.', $output['path']) : $output['path'])
```

Comment:
This isn't quite the correct use for the "code" field. It should indicate the general class or type of error rather than the content of the error itself.

It's possible we just need to use a constant code for all lint messages emitted by this linter. We do that in some of our other linters.

Reply:
Understood, I use path here since the validator doesn't provide error type. I updated it to use linter name (a constant value) like other linters.